src/main/blockchain/staking-service.js

```diff
@@ -0,0 +1,259 @@
+import fetch from 'node-fetch';
+
+import CONFIG from 'common/config';
+import { abi as SELFKEY_ABI } from 'main/assets/data/abi.json';
+import { Token } from '../token/token';
+import BN from 'bignumber.js';
+// import { TxHistory } from './tx-history';
+
+// TODO: use selfkey domain here
+const CONFIG_URL =
+	'https://us-central1-kycchain-master.cloudfunctions.net/airtable?tableName=Contracts';
+
+export class StakingService {
+	constructor({ web3Service }) {
+		this.activeContract = null;
+		this.deprecatedContracts = [];
+		this.web3 = web3Service;
+	}
+	async getStakingInfo(serviceAddress, serviceId, options) {
+		let info = { balance: 0, serviceAddress, serviceId, contract: null, releaseDate: 0 };
+		let contracts = [this.activeContract].concat(this.deprecatedContracts);
+		let balance = 0;
+		options = { ...options };
```

Comment:
@sk91 why are you doing this with options?

Reply:
It is the same as doing "clone" on options. It is a general good practice to make sure it will not change the original 'options' object